src/Entity/User.php

```diff
@@ -53,4 +51,24 @@ public function setGroups($groups): void
     {
         $this->groups = $groups;
     }
+
+    public function addGroup(GroupInterface $group): self
+    {
+        if (!$this->groups->contains($group)) {
+            $this->groups[] = $group;
+            $group->addGroup($this);
```

Comment:
Shouldn't this be `$group->addUser($this)`? We're in the `User` class, right?

Reply:
Bug crushed!